src/server_manager/web_app/gcp_account.ts

```diff
@@ -74,9 +82,9 @@ export class GcpAccount implements gcp.Account {
     for (const response of listInstancesResponses) {
       const instances = response.items ?? [];
       instances.forEach((instance) => {
-        const id = `${this.id}:${instance.id}`;
-        const server = new GcpServer(id, projectId, instance, this.apiClient);
-        result.push(server);
+        const zoneId = instance.zone.substring(instance.zone.lastIndexOf('/') + 1);
+        const locator = {projectId, zoneId, instanceId: instance.id};
+        result.push(new GcpServer(this.id, locator, instance.name, Promise.resolve(), this.apiClient));
```

Comment:
Keep constructing the id here. It's the creator that can ensure unique ids. The instance shouldn't be creating its own id.